code/game/objects/structures/crates_lockers/closets/secure/vault.dm

```diff
@@ -0,0 +1,58 @@
+/obj/structure/closet/secure_closet/vault
+	name = "vault locker"
+	desc = "For when you absolutely need to keep something safe."
+	icon = 'icons/obj/closet.dmi'
+	icon_state = "vault1"
+	density = 1
+	opened = 0
+	large = 1
+	locked = 1
+	anchored = 1 //immovable
+	icon_closed = "vault"
+	icon_locked = "vault1"
+	icon_opened = "vaultopen"
+	icon_broken = "vaultbroken"
+	icon_off = "vaultoff"
+	wall_mounted = 0 //never solid (You can always pass over it)
+	health = 20000
+
+/obj/structure/closet/secure_closet/vault/ex_act(var/severity) //bomb-proof
+	switch(severity)
+		if(1.0)
+			health -= 500
+		if(2.0)
+			health -= 100
+		if(3.0)
+			health -= 20
+	var/list/bombs = search_contents_for(/obj/item/device/transfer_valve)
+	if(!isemptylist(bombs)) //If there's a bomb inside the locker when it's hit with an explosion, the things inside lose their protection
+		for(var/obj/O in src)
+			O.ex_act(severity)
+
+/obj/structure/closet/secure_closet/vault/emp_act(severity) //EMP-proof
+	for(var/obj/O in src)
+		O.emp_act(severity)
+
+/obj/structure/closet/secure_closet/vault/armory
+	name = "\improper Armory vault locker"
+	req_access = list(access_armory)
+
+/obj/structure/closet/secure_closet/vault/armory/lawgiver
```

Comment:
You don't need this if you're not setting vars on it.

You can just define the new() and it'll work.